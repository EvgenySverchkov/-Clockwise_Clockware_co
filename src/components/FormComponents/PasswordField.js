import React, {useContext} from "react";
import PropTypes from "prop-types";

import Tooltip from "../CustomTooltip";
import Context from "../../ContextComponent";

const PasswordField = ({ id }) => {
  const context = useContext(Context);
  return (
  <div className="col-sm-8 col-md-8">
    <Tooltip 
      title={context.warningTooltipMsg} 
      open={context.isOpenWarningTooltip} 
      fieldId = {context.idForTooltip}
      tooltipIsOpen = {context.isOpenWarningTooltip}>
      <input
        id={id}
        type="password"
        name="password"
        className="form-control"
        autoComplete="username"
        required
      />
    </Tooltip>
  </div>
)
}

PasswordField.propTypes = {
  id: PropTypes.string.isRequired,
};

export default PasswordField;
