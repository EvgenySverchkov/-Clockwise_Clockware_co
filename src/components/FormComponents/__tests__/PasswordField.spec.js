import React from 'react';
import {create} from "react-test-renderer";

import PasswordField from "../PasswordField";
import ContextComponent from "../../../ContextComponent";

it("<PasswordField /> test which is used in forms", ()=>{
    const mockProps = {
        id: "1"
    }
    const mockContextValue = {
        warningTooltipMsg: "Message",
        isOpenWarningTooltip: false,
        idForTooltip: 1
    }
    const component = create(
        <ContextComponent.Provider value={mockContextValue}>
            <PasswordField {...mockProps}/>
        </ContextComponent.Provider>
    )
    expect(component.toJSON()).toMatchSnapshot();
});