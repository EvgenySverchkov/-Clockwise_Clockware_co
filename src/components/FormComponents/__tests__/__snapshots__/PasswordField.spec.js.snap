// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordField /> test which is used in forms 1`] = `
<div
  className="col-sm-8 col-md-8"
>
  <input
    aria-describedby={null}
    autoComplete="username"
    className="form-control"
    id="1"
    name="password"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    required={true}
    title="Message"
    type="password"
  />
</div>
`;
